src/cli.rs

```diff
@@ -172,85 +165,98 @@ fn create_cli<'a, 'b>(global_config: &'a GlobalConfig) -> App<'a, 'b> {
-                    ).default_value(default_task_name),
-            ).arg(
-                Arg::with_name("cwd")
-                    .long("--cwd")
-                    .value_name("DIRECTORY")
-                    .help(
-                        "Will set the current working directory. \
-                         The search for the makefile will be from this directory if defined.",
-                    ),
-            ).arg(Arg::with_name("no-workspace").long("--no-workspace").help(
-                "Disable workspace support (tasks are triggered on workspace and not on members)",
-            )).arg(
-                Arg::with_name("no-on-error")
-                    .long("--no-on-error")
-                    .help("Disable on error flow even if defined in config sections"),
-            ).arg(
-                Arg::with_name("envfile")
-                    .long("--env-file")
-                    .value_name("FILE")
-                    .help("Set environment variables from provided file"),
-            ).arg(
-                Arg::with_name("env")
-                    .long("--env")
-                    .short("-e")
-                    .value_name("ENV")
-                    .multiple(true)
-                    .takes_value(true)
-                    .number_of_values(1)
-                    .help("Set environment variables"),
-            ).arg(
-                Arg::from_usage("-l, --loglevel=[LOG LEVEL] 'The log level'")
-                    .possible_values(&["verbose", "info", "error"])
-                    .default_value(default_log_level),
-            ).arg(
-                Arg::with_name("v")
-                    .short("-v")
-                    .long("--verbose")
-                    .help("Sets the log level to verbose (shorthand for --loglevel verbose)"),
-            ).arg(
-                Arg::with_name("experimental")
-                    .long("--experimental")
-                    .help("Allows access unsupported experimental predefined tasks."),
-            ).arg(
-                Arg::with_name("disable-check-for-updates")
-                    .long("--disable-check-for-updates")
-                    .help("Disables the update check during startup"),
-            ).arg(Arg::with_name("print-steps").long("--print-steps").help(
-                "Only prints the steps of the build in the order they will \
-                 be invoked but without invoking them",
-            )).arg(
-                Arg::with_name("list-steps")
-                    .long("--list-all-steps")
-                    .help("Lists all known steps"),
-            ).arg(Arg::with_name("TASK").help("The task name to execute"))
-            .arg(
-                Arg::with_name("TASK_ARGS")
-                    .multiple(true)
-                    .help("Task arguments which can be accessed in the task itself."),
-            ),
-    )
+    App::new("cargo-make")
```

Comment:
haven't tried it, so this would would work for a standalone binary and for a cargo subcommand at the same time?

Reply:
Not 100% sure, I think this change will no longer work with cargo subcommand. In my opinion it is best to choose whether to make this tool a standalone binary or a cargo subcommand rather than support both.

Reply:
don't think for backward compatability we should drop support for old executable and not sure how hard it is to support both.
i'll try it out as well to see complexity

Reply:
ill wait for your decision on whether you want to go with standalone binary, cargo subcommand or both before I spend more time on this